sympy/core/basic.py

```diff
@@ -1525,6 +1525,20 @@ def match(self, pattern, old=False):
         pattern = sympify(pattern)
         return pattern.matches(self, old=old)
 
+
+    def match_between(self, pattern1, pattern2, old=False):
```

Comment:
For prototyping, perhaps new match routines should be added to the rubi module itself. That way, we don't add new API to the core that we might later want to change or remove.